smartfs.lua

```diff
@@ -263,13 +272,27 @@ function smartfs._show_(form, player, params, is_inv)
 	return state
 end
 
+-- Attach a formspec to a node
+function smartfs._attach_nodemeta_(form, nodepos, placer)
+	local state = smartfs._makeState_(form, placer, params, nil, nodepos)
+	if form._reg(state) then
+		state:_attach_nodemeta_()
+	end
+	return state
+end
+
 -- Receive fields from formspec
 local function _sfs_recieve_(state,name,fields)
 	if (fields.quit == "true") then
-		if not state.is_inv then
+		if state.nodepos then
+			state:close() --mark as closed for this user (will be reopened if other user is on the node)
```

Comment:
If multiple users are viewing with the same state, it will prematurely set closed to true

```
closed = true
A opens.
closed = false
B opens
closed = false
A closes
closed = true    <--------------
B closes
closed = true
```

doing reference counting would be better

```
viewers = 0
A opens.
viewers = 1
B opens
viewers = 2
A closes
viewers = 1
B closes
viewers = 0
```

but see main note, I'm not sure sharing state is a good thing